WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/list/BloggingPromptsListFragment.kt

```diff
@@ -0,0 +1,130 @@
+package org.wordpress.android.ui.bloggingprompts.list
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import org.wordpress.android.databinding.BloggingPromptsListFragmentBinding
+import org.wordpress.android.ui.ViewPagerFragment
+import org.wordpress.android.util.extensions.setVisible
+
+class BloggingPromptsListFragment : ViewPagerFragment() {
+
+    private lateinit var binding: BloggingPromptsListFragmentBinding
+    private lateinit var promptsListAdapter: BloggingPromptsListAdapter
+
+    private val parentViewModel: BloggingPromptsListParentViewModel by activityViewModels()
+    private val viewModel: BloggingPromptsListViewModel by viewModels()
+
+    override fun getScrollableViewForUniqueIdProvision(): View = binding.promptsList
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = BloggingPromptsListFragmentBinding.inflate(inflater)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val section = arguments?.getSerializable(LIST_TYPE) as? PromptSection
+        val site = parentViewModel.getSite()
+        initializeViews()
+        setupObservers()
+
+        viewModel.onOpen(site, section)
+    }
+
+    private fun initializeViews() {
+        with(binding.promptsList) {
+            layoutManager = LinearLayoutManager(activity, RecyclerView.VERTICAL, false)
+            addItemDecoration(DividerItemDecoration(context, LinearLayout.VERTICAL))
+            promptsListAdapter = BloggingPromptsListAdapter()
+            adapter = promptsListAdapter
+        }
+    }
+
+    private fun setupObservers(){
+        viewModel.contentViewState.observe(viewLifecycleOwner) {
+            setContentViewVisibility(
+                    it.isVisible,
+                    it.list
+            )
+        }
+        viewModel.errorViewState.observe(viewLifecycleOwner) {
+            setErrorViewVisibility(
+                    it.isVisible,
+                    it.imageResId,
+                    it.titleTextResId,
+                    it.subtitleTextResId,
+                    it.buttonTextResId
+            )
+        }
```

Comment:
I feel like contentViewState and errorViewState could be merged into a single UiState (that may be composed by those 2 state in principle). An advantage that I see is also that it seems you need to set value on both contentViewState and errorViewState all over the places. Merging them could result in simpler code. wdyt?

Reply:
Yes. Maybe I misunderstood something or failed to explain my intentions clearly. But that was the original intent behind [this approach](https://github.com/metalwihen/WordPress-Android/pull/8#discussion_r996455509). 

That said, let me think about how to go about it with our current approach of testing the visibility flags in the viewmodel as well as representing both states as a single ui state. On the off chance I might misinterpret your intentions, do you have a reference in mind I could use?

Reply:
I think that's good we moved the state to the ViewModel for the testability criteria we were already saying. Also, I think it's ok to use sealed classes to model the content and error states as you did. What I was suggesting is avoiding having 2 observers (that you are anyways calling both all over the places) and have only 1 UiState data class (that in principle can still contain instances of the sealed classes you used to model content and error state). Looking for UiState in the codebase could possibly help.

Reply:
Done.